File: src/main/java/org/example/bankramenserver/infrastructure/integration/IntegrationConnector.java
package org.example.bankramenserver.infrastructure.integration;

import org.example.bankramenserver.infrastructure.integration.domain.IntegrationConnectionProperties;
import org.example.bankramenserver.infrastructure.integration.domain.IntegrationOutbox;

public interface IntegrationConnector {

    String connectorType();

    void dispatch(IntegrationOutbox outbox, IntegrationConnectionProperties.Connection connection) throws Exception;
}

File: src/main/kotlin/org/example/bankramenserver/infrastructure/integration/IntegrationConnector.kt
package org.example.bankramenserver.infrastructure.integration

import org.example.bankramenserver.infrastructure.integration.domain.IntegrationConnectionProperties
import org.example.bankramenserver.infrastructure.integration.domain.IntegrationOutbox

interface IntegrationConnector {
    fun connectorType(): String

    @Throws(Exception::class)
    fun dispatch(outbox: IntegrationOutbox, connection: IntegrationConnectionProperties.Connection)
}
